docs/cookbook/using-docker.md

```diff
@@ -0,0 +1,343 @@
+# Building Docker images for Apostrophe projects
+
+[Docker](https://www.docker.com/) is a containerization platform that lets developers build an image for their projects and then run it anywhere. This document will guide you through setting up Docker and using it to create an image for hosting your Apostrophe project.
+
+The initial steps of this guide will assume that you will be hosting your database, project, and uploaded assets on the same server. The second part will outline steps for configuring to use the AWS S3 service for hosting assets. Finally, the third portion will provide guidance for using the MongoDB Atlas multi-cloud database.
+
+## Creating the image
+### Install Docker
+
+Docker can be installed on Mac, Windows, and Linux machines with either a CLI interface or using the Docker Desktop application, which acts as a graphical interface to the Docker engine. For this tutorial, we will use the CLI version. You can read more about Docker and install it on your machine by following the instructions in the Docker [docs](https://docs.docker.com/get-started/). Feel free to walk through the tutorials that you find there, but it isn't necessary before following this tutorial. We will cover any necessary terminology as we walk step-by-step through getting Apostrophe running. However, please verify that your Docker install works before continuing.
+
+### Apostrophe project setup
+
+For this tutorial, we will be using the [a3-demo](https://github.com/apostrophecms/a3-demo) template. However, you can also use an existing project or create a new one by following our getting started [tutorial](https://v3.docs.apostrophecms.org/guide/setting-up.html). If using the a3-demo, follow the link and click on the "Use this template" button to fork the template into your own repo. Next, clone the repo to your local machine and open it in your favorite code editor.
+
+### Creating the dockerfile
+
+The [`Dockerfile`](https://docs.docker.com/engine/reference/builder/) passes a series of command line instructions to build an image. Below is an example of a file to build a container for a basic Apostrophe project:
+
+<AposCodeBlock>
+
+```bash
+FROM node:lts-alpine3.15
+
+WORKDIR /srv/www/apostrophe
+
+RUN chown -R node: /srv/www/apostrophe
+USER node
+
+COPY --chown=node package*.json /srv/www/apostrophe/
+
+NODE_ENV=production
+RUN npm ci
+
+COPY --chown=node . /srv/www/apostrophe/
+
+RUN ./scripts/build-assets.sh
+
+CMD ["node", "app.js"]
+```
+<template v-slot:caption>
+  Dockerfile
+</template>
+
+</AposCodeBlock>
+
+Let's walk briefly through each of the lines. The first line specifies that this image will extend an existing image with the long-term support version of node.js running on an alpine linux v3.15 operating system. If you need another version of node or a newer version of alpine linux, you should alter this line to build from an official node.js [image](https://hub.docker.com/_/node/) or a 3<sup>rd</sup> party image.
```

Comment:
They don't have to use alpine at all, so I'd say "if you need another version of node, another version of Alpine Linux, or prefer not to use Alpine Linux as your container OS"

Reply:
The stock nodejs images are Ubuntu based

Reply:
Altered.